### ISSUE TITLE
Players get kicked when a plugin calls `Player#batchPacketData`

### ISSUE THREAD

**Issue by roridev** [PowerNukkit/PowerNukkit#887]
Plugins Affected
--- 
- MobPlugin

Relevant Info
--- 
 
 IMHO this a bit is more of a server issue than a plugin. I found out that the problem > comes from [this](https://github.com/Nukkit-coders/MobPlugin/blob/04ac8134b0f1db39675c31af01f93521c3c53104/src/main/java/nukkitcoders/mobplugin/entities/BaseEntity.java#L477-L479) piece of code. Instead of using broadcast method which encodes packet before putting it to the interface (so that why PN itself has no problem), plugin iterates players and sends packet using Player#batchDataPacket() method. Concurrency issues are not really expected to occur when using such code.

_Originally posted by @AntonPPDS in https://github.com/PowerNukkit/PowerNukkit/issues/858#issuecomment-755505643_

**Comment by MrRazamataz** [comment]
https://hastebin.com/uqiruciloq
This includes:

- [ ]  Advanced-Welcome
- [ ] BanSystemNK
- [ ] ChunkReset
- [ ] DbLib
- [ ] TheEnd
- [ ] EssentialsNK
- [ ] FastAsyncWorldEdit
- [ ] LlamaButcher
- [ ] MultiWorld
- [ ] RazaOnePlayerSleep
- [ ] Pokkit
- [ ] VanillaCommandBlock
- [ ] FormAPI
- [ ] NPC

**Comment by AntonPPDS** [comment]
The issue seems to be fixed here 718dd63. But there are possible compatibility issue when using the same packet instance to send data packets with different parameters by encoding it again. If two or more such packets are sent in the same RakNet tick, all of them will have only the last one's buffer.
```java
SomePacket pk = new SomePacket();

pk.param = 1;
pk.encode();
player.dataPacket(pk);

pk.param = 2;
pk.encode();
player.dataPacket(pk);
```

**Comment by joserobjr** [comment]
Closing as stale because this issue seems to be fixed but it is hard to know exactly when the fix happened. If the issue persists, please reopen with a new debugpaste.
